package/MDAnalysis/analysis/base.py

```diff
@@ -40,17 +40,71 @@
 logger = logging.getLogger(__name__)
 
 
+class _Results:
+    r"""Class storing results obtained from an analysis.
+
+    The class stores all results obatined from
+    an analysis after the ``run`` call.
+    """
+
+    def __init__(self, analysis_cls):
+        """
+        Parameters
+        ----------
+        analysis_cls : AnalysisClass
+            parent analysis class
+        """
+        self.analysis_cls = analysis_cls
+
+    def __len__(self):
+        # Subract the analysis class attribute
+        return len(self.__dict__) - 1
+
+    def get_attribute_list(self):
+        """A list of all result attribute names."""
+        return [key for key in self.__dict__.keys() if key != "analysis_cls"]
+
+    def __repr__(self):
+        analysis_name = type(self.analysis_cls).__name__
+        return f"<{analysis_name} results with {len(self)}" \
+               f" attribute{'s'[len(self) == 1:]}>"
+
+    def __str__(self):
+        analysis_name = type(self.analysis_cls).__name__
+        attribute_list = self.get_attribute_list()
+        str_repr = f"<{analysis_name} results with "
+        str_repr += f"attribute{'s'[len(self) == 1:]}: "
+
+        if len(self) <= 10:
+            str_repr += ", ".join(attribute_list)
+        else:
+            str_repr += ", ".join(attribute_list[:3])
+            str_repr += " ... " + ", ".join(attribute_list[-3:])
+
+        return str_repr + ">"
+
+    def __eq__(self, other):
+        if self.__dict__.keys() != other.__dict__.keys() :
+            raise TypeError("Can't compare results from with different"
+                            " attributes.")
+        return self.__dict__ == other.__dict__
+
+    def __ne__(self, other):
```

Comment:
Defining `__ne__` as this here seems unneeded, as the [datamodel](https://docs.python.org/3/reference/datamodel.html#object.__ne__) already delegates it by default to be the negation of the result of `__eq__`.

Reply:
Done.